Added file name data to collection items

rgladwell opened the pull request:
Adds a `file` property to the collection item's data which contains the file name of the generated file. For example, this can be used in templates to create links.

Includes test case.

lambtron commented:
Thanks for submitting the PR! Can this be achieved by using a second plugin (metalsmith-path or metalsmith-paths) after collections that adds the `.file` property?

@dominicbarnes thoughts here?

rgladwell commented:
@lambtron thanks, using metalsmith-path seemed to work. Maybe we should document this somewhere in this project though?

dominicbarnes commented on lib/index.js:
```diff
@@ -5,6 +5,7 @@ var Matcher = require('minimatch').Minimatch;
 var unique = require('uniq');
 var read = require('fs').readFileSync;
 var loadMetadata = require('read-metadata').sync;
+var path = require('path');
```
This is superfluous, looks like it was added and left behind when not needed anymore.

rgladwell commented:
Hey @dominicbarnes, good spot. I've rebased a couple of updates including a fix for the above:

  * Removed un-used module import
  * Changed name of property file -> path

rgladwell commented:
@lambtron thanks, when do you think you'll release a new version of metalsmith-collections?

bashaus commented:
This is causing conflicts with metalsmith-path and metalsmith-paths, see https://github.com/segmentio/metalsmith-collections/pull/62
